### PR TITLE
docs: add Thrift protocol test suite design

### PR DESCRIPTION
**Pull request by eric-wang-1990**
Add comprehensive design document for ADBC driver Thrift protocol test suite.

Key features:
- Language-agnostic test specifications (~300 test cases across 16 categories)
- Standalone Go proxy server for failure injection testing
- Multi-language support strategy (C#, Java, C++, Go)
- Extractable design for future common repository

Initial implementation targets C# ADBC driver with plans to extend to:
- Java (JDBC driver)
- C++ (ODBC driver)
- Go (ADBC driver)

The design enables comprehensive testing of:
- Session lifecycle and management
- Statement execution (sync/async)
- Metadata operations
- Arrow format and compression
- CloudFetch results and failure scenarios
- Parameterized queries
- Error handling and recovery
- Concurrency and edge cases

Directory structure:
- docs/designs/thrift-protocol-tests/ - Design and specifications
- test-infrastructure/proxy-server/ - Standalone proxy (to be implemented)

Related to runtime Thrift tests and ADBC E2E tests, but focuses on comprehensive protocol compliance and failure scenario testing.

## What's Changed

Please fill in a description of the changes here.

**This contains breaking changes.**  

Closes #NNN.

### CODE REVIEW

File: test-infrastructure/README.md [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+    assertions:
+      - type: query_succeeds
+      - type: thrift_call_count
+        method: FetchResults
+        expected: baseline_count + 1
+```
+
+See [specs/README.md](./specs/README.md) for full documentation.
+
+**Planned Test Suites (~300 test cases across 16 categories):**
```
Comment (eric-wang-1990) [inline test-infrastructure/README.md]: This seem can be removed since we already have detailed tests in the specs/README.md

File: test-infrastructure/specs/README.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+| `schema_valid` | Result schema is valid | - |
+| `batch_not_empty` | First batch has data | - |
+| `thrift_call_count` | Verify Thrift method call count | `method`, `expected` |
+| `cloud_download_count` | Verify CloudFetch download count | `expected` |
+
+## Measurement Types
+
+| Type | Description | Save As |
+|------|-------------|---------|
+| `thrift_method` | Count calls to Thrift method | Variable name |
```
Comment (eric-wang-1990) [inline test-infrastructure/specs/README.md]: The thrift_method should have a name as well like GetCatalogs, etc.
